routes/index.js

```diff
@@ -1,81 +1,102 @@
+    name: req.sanitize(req.body.name)
+  };
+  if (req.body.club != "") {
+    data["club"] = req.sanitize(req.body.club);
+  }
+  if (req.body.long_desc != "") {
+    data["long_desc"] = req.sanitize(req.body.long_desc);
+  }
+
+  let status = { task: "complete" };
+  console.log(data);
+  IDEA.create(data, (err, created) => {
+    if (err) {
+      status.task = "error";
+      throw err;
     }
-    if(req.body.club != ""){
-        data["club"] = req.sanitize(req.body.club);
+    if (!created) status.task = "NOT OK";
+    res.json(status);
+  });
+});
+
+//to fetch list of idea from db
+router.get("/idea/list/", (req, res) => {
+  let response = { status: "OK", list: [] };
+  IDEA.find({}, "name title desc ideaId", (err, ideaList) => {
+    if (err) {
+      response.status = "ERROR";
+      throw err;
     }
-    if(req.body.long_desc != ""){
-        data["long_desc"] = req.sanitize(req.body.long_desc);
+    if (ideaList) {
+      response.list = ideaList;
     }
-    
-    let status = {task: "complete"};
-    console.log(data);
-    IDEA.create(data, (err, created)=>{
-       if (err){ 
-            status.task = "error";
-            throw err;
-       }if (!created)
-            status.task = "NOT OK";
-        res.json(status);
-    });
-    
+    res.json(response);
+  });
 });
 
-//to fetch list of idea from db
-router.get("/idea/list/", (req,res)=>{
-    let response = {"status": "OK", list: []};
-    IDEA.find({}, "name title desc ideaId", (err, ideaList)=>{
-       if  (err)
-       {
-           response.status = "ERROR";
-           throw err;
-       }if (ideaList)
-       {
-           response.list = ideaList;
-       }
-       res.json(response);
-    });
+//to get detail of any idea
+router.get("/idea/:ideaId/", (req, res) => {
+  let response = { status: "OK", idea: {} };
+
+  const ideaID = req.params.ideaId;
+
+  IDEA.find({ ideaId: ideaID }, (err, ideaGet) => {
+    if (err) {
+      response.status = "ERROR";
+      throw err;
+    }
+    if (ideaGet) {
+      ideaGet._id = " ";
+      response.idea = ideaGet;
+    }
+    res.json(response);
+  });
 });
 
-//to get detail of any idea 
-router.get("/idea/:ideaId/", (req,res)=>{
-    let response = {"status": "OK", idea: {}};
-    
-    const ideaID  = req.params.ideaId;
-    
-    IDEA.find({ideaId: ideaID} ,(err, ideaGet)=>{
-       if  (err)
-       {
-           response.status = "ERROR";
-           throw err;
-       }if (ideaGet){
-            ideaGet._id  = " ";
-            response.idea = ideaGet;
-       }
-        res.json(response);
-    }); 
-    
-});  
-       
+// Like any idea
+router.get("/like/idea", (req, res) => {
+  let { ideaId } = req.query;
+  let ip = requestIp.getClientIp(req);
+  IDEA.findByIdAndUpdate(
+    ideaId,
+    {
+      $addToSet: {
+        likes: ip
+      }
+    },
+    { safe: true, new: true }
+  )
+    .then(response => {
+      return res.send(response);
+    })
+    .catch(e => {
+      return res.send({
```

Comment:
- Please use res.status(500).json(<object>) as by default res.send() set http status to 200.